accelerator/proxy_apis.c

```diff
@@ -313,46 +321,52 @@ status_t api_get_trytes(const iota_client_service_t* const service, const char*
   return ret;
 }
 
-status_t api_remove_neighbors(const iota_client_service_t* const service, const char* const obj, char** json_result) {
+status_t api_proxy_apis(const iota_client_service_t* const service, const char* const obj, char** json_result) {
   status_t ret = SC_OK;
-  remove_neighbors_req_t* req = remove_neighbors_req_new();
-  remove_neighbors_res_t* res = remove_neighbors_res_new();
-  char_buffer_t* res_buff = char_buffer_new();
-  if (req == NULL || res == NULL || res_buff == NULL) {
-    ret = SC_TA_OOM;
-    ta_log_error("%s\n", "SC_TA_OOM");
-    goto done;
-  }
-
-  lock_handle_lock(&cjson_lock);
-  if (service->serializer.vtable.remove_neighbors_deserialize_request(obj, req) != RC_OK) {
-    lock_handle_unlock(&cjson_lock);
-    ret = SC_CCLIENT_JSON_PARSE;
-    ta_log_error("%s\n", "SC_CCLIENT_JSON_PARSE");
-    goto done;
-  }
-  lock_handle_unlock(&cjson_lock);
-
-  lock_handle_lock(&cjson_lock);
-  if (iota_client_remove_neighbors(service, req, res) != RC_OK) {
-    lock_handle_unlock(&cjson_lock);
-    ret = SC_CCLIENT_FAILED_RESPONSE;
-    ta_log_error("%s\n", "SC_CCLIENT_FAILED_RESPONSE");
-    goto done;
+  const uint8_t max_cmd_len = 30;
+  char command[max_cmd_len];
+
+  proxy_apis_command_req_deserialize(obj, command);
+  // TODO more proxy APIs should be implemented
+  enum {
+    djb2__checkConsistency = 6052213440223724181U,
+    djb2__findTransactions = 10456093532624710815U,
+    djb2__getBalances = 13875264399672181694U,
+    djb2__getInclusionStates = 14323307694851473837U,
+    djb2__getNodeInfo = 13875264928864391863U,
+    djb2__getTrytes = 249889515354659152U
+  };
+
+  switch (hash_algo_djb2(command)) {
```

Comment:
With the aid of hash function and jump table conjuncted with `switch`, the dispatch is now deterministic, that implies constant time manipulation for each proxy API request.

Reply:
Add comments for explaining the considerations of introducing DJB2 hash functions: reduce the amount of string comparisons and deterministic execution time.